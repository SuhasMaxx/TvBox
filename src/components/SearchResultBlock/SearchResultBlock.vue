<template src="./searchresultblock.template.html"></template>
<script>
import { UtilsMixins } from "../../utils/utils.mixin.js";
import TvBadges from "../TvBadges/TvBadges.vue";
import NoImage from "../NoImage/NoImage.vue";
/**
 * Small show block with show info on Home page
 *
 * @class ShowBlock
 */
export default {
  components: {
    TvBadges,
    NoImage,
  },
  mixins: [UtilsMixins],
  props: {
    showDetails: {
      type: Object,
      default: null,
    },
  },
  computed: {
    premierYear: function () {
      return this.getYear(this.showDetails.premiered);
    },
  },
  methods: {
    gotoShowDetails() {
      this.$router.push("/show/" + this.showDetails.id);
    },
  },
};
</script>
<style src="./searchresultblock.style.css" scoped></style>
