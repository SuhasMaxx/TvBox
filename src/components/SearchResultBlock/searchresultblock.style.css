.search-result-block{
    /* padding: 1em; */
    margin: 1em 0 1em 1em;
    /* border: 2px solid grey; */
    max-width: 500px;
    display: flex;
    flex-flow: row;
    color: var(--rich-black-fogra-39);
    /* border-radius: 5px;
    background-color: var(--light-gray);
    box-shadow: 0px 0px 10px 2px var(--silver-chalice); */
    cursor: pointer;
}

.show-block { transition: all .2s ease-in-out; }
.show-block:hover { transform: scale(1.1); }

.show-block:hover{
    box-shadow: 0px 0px 10px 2px var(--silver-chalice);
}

h2{
    color: white;
    padding: 5px;
}

.show-img{
    height: 180px;
    width: 120px;
}

.show-image{

}

.show-details{

}