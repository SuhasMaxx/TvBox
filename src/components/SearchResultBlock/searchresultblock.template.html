<div class="search-result-block" @click="gotoShowDetails()">
    <div class="show-image">
        <img v-if="showDetails.image" :src="showDetails.image.medium" class="show-img" :title="showDetails.name"/>
        <no-image v-if="!showDetails.image"></no-image>
    </div>
    <div class="show-details">
        <h2>{{showDetails.name}}</h2>
        <tv-badges :showDetails="showDetails"></tv-badges>
        <div v-if="showDetails.network"><b>Network:</b> {{showDetails.network.name}} ({{showDetails.network.country.code}})</div>
        <div class="genres" v-if="showDetails.genres?.length">
            <b>Genre: </b>
            <span v-for="(genre,index) in showDetails.genres">
                {{genre}}
                <span v-if="index+1 < showDetails.genres.length">&nbsp;|&nbsp;</span>
            </span>
        </div>
        <div v-if="showDetails.type"><b>Type: </b>{{showDetails.type}}</div>
    </div>
</div>