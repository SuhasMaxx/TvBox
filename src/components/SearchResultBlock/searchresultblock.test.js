import SearchResultBlock from './SearchResultBlock.vue';
import { shallowMount } from '@vue/test-utils';
import showDetails from '../../utils/showDetails.mock.js';

describe('SearchResultBlock.vue', () => {
  it('render the component', () => {
    const wrapper = shallowMount(SearchResultBlock,{
        props: {
            showDetails: showDetails.singleShowDetails
        }
    });
    expect(wrapper.element).toMatchSnapshot()
  })
})