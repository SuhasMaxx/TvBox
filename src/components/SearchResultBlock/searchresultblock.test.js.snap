// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultBlock.vue render the component 1`] = `
<div
  class="search-result-block"
>
  <div
    class="show-image"
  >
    <img
      class="show-img"
      src="https://static.tvmaze.com/uploads/images/medium_portrait/69/174906.jpg"
      title="Grimm"
    />
    <!--v-if-->
  </div>
  <div
    class="show-details"
  >
    <h2>
      Grimm
    </h2>
    <tv-badges-stub
      showdetails="[object Object]"
    />
    <div>
      <b>
        Network:
      </b>
       NBC (US) 
    </div>
    <div
      class="genres"
    >
      <b>
        Genre: 
      </b>
      
      <span>
        Drama 
        <span>
           | 
        </span>
      </span>
      <span>
        Crime 
        <span>
           | 
        </span>
      </span>
      <span>
        Supernatural 
        <!--v-if-->
      </span>
      
    </div>
    <div>
      <b>
        Type: 
      </b>
      Scripted
    </div>
  </div>
</div>
`;
